drivers/adc/Kconfig.mcux

```diff
@@ -40,6 +40,7 @@ config ADC_MCUX_LPADC
 config ADC_MCUX_ETC
 	bool "MCUX ADC ETC driver"
 	depends on HAS_MCUX_ADC_ETC
+	depends on ADC_MCUX_LPADC
```

Comment:
Why would this depend on the LPADC? Parts like the RT1064 which do not have an LPADC have the ADC_ETC IP.

Reply:
What about `depends on ADC`?  That needs the ADC driver included, but does not depend on specific IP.

Reply:
I think that is ok. I'll also note that the ADC_ETC IP isn't really supported in Zephyr right now- all this Kconfig does is enables a user to include the HAL driver in their application.

Reply:
I updated the dependency. if any ADC IP is defined, ETC is defined.